### ISSUE TITLE
Consider adding debug points for boolean logic

### ISSUE THREAD

**Issue by dsyme** [dotnet/fsharp#11980]
Complex boolean logic is quite common in F#, e.g. see examples below.  It is really hard to debug.

I propose we add debug points on left and right of `a && b` and `a || b` so you can step through this logic and hit breakpoints.

C# doesn't allow breaking in boolean logic or any expression.   However 
1. F# already steps through any `match` and `if ... then .. else` expressions so adding this seems natural enough.
2. I believe complex boolean logic is more common in F# than C# because of the expression-oriented nature of the language

A possible downside is that of "too much stepping" so it takes too long to step through a function.  However I've been frustrated by the lack of debugging for boolean logic enough that I think we should prefer the addditional debug points.

Examples

```fsharp
let isRecdOrStructTyconRefAssumedImmutable (g: TcGlobals) (tcref: TyconRef) =
    tcref.CanDeref &&
    not (isRecdOrUnionOrStructTyconRefDefinitelyMutable tcref) ||
    tyconRefEq g tcref g.decimal_tcr ||
    tyconRefEq g tcref g.date_tcr
```

or 

![MicrosoftTeams-image](https://user-images.githubusercontent.com/7204669/129640456-a5f3ed9a-19a4-4945-ad17-b4cfb591f3d8.png)

**Comment by zanaptak** [comment]
Better too many steps than too few, they at least can be mitigated with Run to Cursor or setting a later breakpoint.

**Comment by dsyme** [comment]
This is being implemented #12432

**Comment by dsyme** [comment]
CLosing as completed in #12432
